examples/dapperish.go

```diff
@@ -19,10 +18,10 @@ import (
 func client() {
 	reader := bufio.NewReader(os.Stdin)
 	for {
-		span := opentracing.StartTrace("getInput")
-		ctx := opentracing.BackgroundGoContextWithSpan(span)
+		ctx, span := opentracing.BackgroundContextWithSpan(
+			opentracing.StartTrace("getInput"))
 		// Make sure that global trace tag propagation works.
-		span.TraceContext().SetTraceAttribute("User", os.Getenv("USER"))
+		span.SetTraceAttribute("User", os.Getenv("USER"))
```

Comment:
don't mean to distract the issue, but I will :package: 

I *really* like Tag Sets where propagation is simply a property of a tag. The "Trace" part of the name "TraceAttribute" has resulted in heaps of discussion, notably as no-one can guess by its name that it a propagation tag, but also the common "is it a part of the trace?" Ex. in zipkin trace id, id, parent id, and sampled decision are "trace attributes", ie only one field has anything to do with the trace. 

I really would rather we be able to deconflate this whole thing by saying `Span.propagateTag("User", os.Getenv("USER"))`, or worse but still better `Span.propagateAttribute("User", os.Getenv("USER"))`. I have consistently found TraceAttribute an unnecessarily distracting term.